cpp/tests/test_abm_location.cpp

```diff
@@ -85,50 +87,52 @@ TEST(TestLocation, CacheExposureRate)
 {
     using testing::Return;
 
-    {
-        mio::abm::AgeGroup age =
-            mio::abm::AgeGroup(mio::UniformIntDistribution<int>()(0, int(mio::abm::AgeGroup::Count) - 1));
-        mio::abm::VirusVariant variant =
-            mio::abm::VirusVariant(mio::UniformIntDistribution<int>()(0, int(mio::abm::VirusVariant::Count) - 1));
-
-        auto t  = mio::abm::TimePoint(0);
-        auto dt = mio::abm::seconds(10000);
-
-        mio::abm::GlobalInfectionParameters params;
-
-        // setup a location with some chance of exposure
-        auto home      = mio::abm::Location(mio::abm::LocationType::Home, 0, 1);
-        auto location  = mio::abm::Location(mio::abm::LocationType::PublicTransport, 0, 3);
-        auto infected1 = mio::abm::Person(home, age);
-        infected1.add_new_infection(mio::abm::Infection(variant, age, params, t));
-        infected1.migrate_to(location, {0});
-        auto infected2 = mio::abm::Person(home, age);
-        infected2.add_new_infection(mio::abm::Infection(variant, age, params, t));
-        infected2.migrate_to(location, {0, 1});
-
-        //cache precomputed results
-        location.cache_exposure_rates(t, dt);
-
-        EXPECT_NEAR((location.get_cells()[0].m_cached_exposure_rate_contacts[{variant, age}]), 0.0020454543682044937,
-                    1e-14);
-        EXPECT_NEAR((location.get_cells()[0].m_cached_exposure_rate_air[{variant}]), 0.0020454543682044937, 1e-14);
-        EXPECT_NEAR((location.get_cells()[1].m_cached_exposure_rate_contacts[{variant, age}]), 0.0010227271841022469,
-                    1e-14);
-        EXPECT_NEAR((location.get_cells()[1].m_cached_exposure_rate_air[{variant}]), 0.0010227271841022469, 1e-14);
-        EXPECT_NEAR((location.get_cells()[2].m_cached_exposure_rate_contacts[{variant, age}]), 0, 1e-14);
-        EXPECT_NEAR((location.get_cells()[2].m_cached_exposure_rate_air[{variant}]), 0, 1e-14);
-
-        // should also work with capacities
-        location.set_capacity_adapted_transmission_risk_flag(true);
-        location.set_capacity(2, 22, 0); // Capacity for Cell 1
-        location.set_capacity(2, 22, 1); // Capacity for Cell 2
-        location.set_capacity(2, 22, 2); // Capacity for Cell 3
-        location.cache_exposure_rates(t, dt);
-
-        EXPECT_NEAR((location.get_cells()[0].m_cached_exposure_rate_air[{variant}]), 0.0061363631046134817, 1e-14);
-        EXPECT_NEAR((location.get_cells()[1].m_cached_exposure_rate_air[{variant}]), 0.0030681815523067408, 1e-14);
-        EXPECT_NEAR((location.get_cells()[2].m_cached_exposure_rate_air[{variant}]), 0, 1e-14);
-    }
+    auto rng = mio::RandomNumberGenerator();
+
+    mio::abm::AgeGroup age = mio::abm::AgeGroup(
+        mio::UniformIntDistribution<int>::get_instance()(rng, 0, int(mio::abm::AgeGroup::Count) - 1));
```

Comment:
There is no mocking here, we are only using the RNG to randomize parameters that SHOULD have no influence on the test. Because there is definitely no mocks in place, we could just use the constructor to create a new distribution object instead of using the distribution from `get_instance`. But now I made the constructor of the distributions private for safety, because we want to make sure that `get_instance` is used during the simulation to enable mocking. Because the constructor is private, we need to use `get_instance` here as well, even though it's not necessary for mocking.